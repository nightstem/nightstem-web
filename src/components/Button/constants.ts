export const BUTTON_SHAPE = {
  SQUARE: 'square',
  ROUNDED: 'rounded',
  CIRCLE: 'circle',
} as const;

export const BUTTON_VARIANT = {
  SOLID: 'solid',
  OUTLINED: 'outlined',
  GHOST: 'ghost',
  TEXT: 'text',
} as const;

export const BUTTON_SIZE = {
  XS: 'xs',
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
  XL: 'xl',
} as const;

export const BUTTON_COLORS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  NEUTRAL: 'neutral',
} as const;
