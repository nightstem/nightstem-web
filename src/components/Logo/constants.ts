export const LOGO_MODES = {
  ICON: 'ICON',
  HORIZONTAL: 'HORIZONTAL',
  FULL: 'FULL',
} as const;

export const LOGO_SIZES = {
  DISPLAY: 'display',
  XXL: 'xxl',
  XL: 'xl',
  LG: 'lg',
  MD: 'md',
  SM: 'sm',
  BODY: 'body',
  CAPTION: 'caption',
  HINT: 'hint',
} as const;

export const LOGO_ANIMATIONS = {
  NONE: 'none',
  PULSE: 'pulse',
} as const;

export const LOGO_ANIMATION_CONFIG = {
  [LOGO_ANIMATIONS.NONE]: '',
  [LOGO_ANIMATIONS.PULSE]: 'animate-pulse',
} as const;

export const LOGO_SIZE_CONFIG = {
  [LOGO_SIZES.DISPLAY]: {
    textClass: 'heading-display',
    logoClass: 'h-20 w-20', // ~76px for 5.61rem
    gapClass: 'gap-4',
  },
  [LOGO_SIZES.XXL]: {
    textClass: 'heading-xxl',
    logoClass: 'h-14 w-14', // ~57px for 4.209rem
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.XL]: {
    textClass: 'heading-xl',
    logoClass: 'h-11 w-11', // ~43px for 3.157rem
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.LG]: {
    textClass: 'heading-lg',
    logoClass: 'h-8 w-8', // 32px for 2.369rem (matches MaintenanceMode)
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.MD]: {
    textClass: 'heading-md',
    logoClass: 'h-6 w-6', // ~24px for 1.777rem
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.SM]: {
    textClass: 'heading-sm',
    logoClass: 'h-[18px] w-[18px]', // ~18px for 1.333rem
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.BODY]: {
    textClass: 'text-body',
    logoClass: 'h-[14px] w-[14px]', // ~14px for 1rem
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.CAPTION]: {
    textClass: 'text-caption',
    logoClass: 'h-[10px] w-[10px]', // ~10px for 0.75rem
    gapClass: 'gap-1',
  },
  [LOGO_SIZES.HINT]: {
    textClass: 'text-hint',
    logoClass: 'h-2 w-2', // ~8px for 0.563rem
    gapClass: 'gap-1',
  },
} as const;
