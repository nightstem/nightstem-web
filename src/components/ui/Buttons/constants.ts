export const BUTTON_SHAPE = {
  SQUARE: 'square',
  CIRCLE: 'circle',
} as const;

export const BUTTON_VARIANT = {
  SOLID: 'solid',
  OUTLINED: 'outlined',
  GHOST: 'ghost',
  TEXT: 'text',
} as const;

export const BUTTON_SIZE = {
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
} as const;

export const BUTTON_COLORS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  NEUTRAL: 'neutral',
} as const;
