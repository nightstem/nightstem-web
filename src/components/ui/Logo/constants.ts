export const LOGO_MODES = {
  ICON: 'icon',
  HORIZONTAL: 'horizontal',
  FULL: 'full',
} as const;

export const LOGO_SIZES = {
  DISPLAY: 'display',
  XXL: 'xxl',
  XL: 'xl',
  LG: 'lg',
  MD: 'md',
  SM: 'sm',
  BODY: 'body',
  CAPTION: 'caption',
  HINT: 'hint',
} as const;

export const LOGO_ANIMATIONS = {
  NONE: 'none',
  PULSE: 'pulse',
} as const;

export const LOGO_ANIMATION_CONFIG = {
  [LOGO_ANIMATIONS.NONE]: '',
  [LOGO_ANIMATIONS.PULSE]: 'animate-pulse',
} as const;

export const LOGO_SIZE_CONFIG = {
  [LOGO_SIZES.DISPLAY]: {
    textClass: 'heading-display', // 5.61rem
    logoClass: 'h-[4.488rem] w-[4.488rem]', // 5.61 * 0.8
    gapClass: 'gap-4',
  },
  [LOGO_SIZES.XXL]: {
    textClass: 'heading-xxl', // 4.209rem
    logoClass: 'h-[3.367rem] w-[3.367rem]', // 4.209 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.XL]: {
    textClass: 'heading-xl', // 3.157rem
    logoClass: 'h-[2.526rem] w-[2.526rem]', // 3.157 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.LG]: {
    textClass: 'heading-lg', // 2.369rem
    logoClass: 'h-[1.895rem] w-[1.895rem]', // 2.369 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.MD]: {
    textClass: 'heading-md', // 1.777rem
    logoClass: 'h-[1.422rem] w-[1.422rem]', // 1.777 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.SM]: {
    textClass: 'heading-sm', // 1.333rem
    logoClass: 'h-[1.066rem] w-[1.066rem]', // 1.333 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.BODY]: {
    textClass: 'text-body', // 1rem
    logoClass: 'h-[0.8rem] w-[0.8rem]', // 1 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.CAPTION]: {
    textClass: 'text-caption', // 0.75rem
    logoClass: 'h-[0.6rem] w-[0.6rem]', // 0.75 * 0.8
    gapClass: 'gap-1',
  },
  [LOGO_SIZES.HINT]: {
    textClass: 'text-hint', // 0.563rem
    logoClass: 'h-[0.45rem] w-[0.45rem]', // 0.563 * 0.8
    gapClass: 'gap-1',
  },
} as const;
