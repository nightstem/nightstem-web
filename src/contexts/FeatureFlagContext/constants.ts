export const FEATURE_FLAGS = {
  'maintenance:global': false,
} as const;

export type FeatureFlagKey = keyof typeof FEATURE_FLAGS;

export type FeatureFlagValues = {
  [K in FeatureFlagKey]: (typeof FEATURE_FLAGS)[K];
};
