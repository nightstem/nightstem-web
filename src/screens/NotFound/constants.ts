export type NotFoundPhrase = {
  title: string;
  description: string;
};

export const defaultNotFound: NotFoundPhrase = {
  title: 'Page not found',
  description: "Let's head back home.",
};

export const notFoundList: NotFoundPhrase[] = [
  // Programming Humor (playful, not real errors)
  {
    title: 'Function? What Function?',
    description: 'We looked for the handler and found a hammock.',
  },
  {
    title: 'Undefined-ish Page',
    description: "There's a value here, but it's purely philosophical.",
  },
  {
    title: 'Breakpoint on Empty Route',
    description: 'Paused execution; nothing worth stepping into.',
  },
  {
    title: "console.log('Nope')",
    description: 'Output: silence, sprinkled with crickets.',
  },
  {
    title: 'Hello, World? Hello?',
    description: 'No response. Try waving at the homepage.',
  },

  // Magic / Fantasy
  {
    title: 'Spell Fizzled',
    description: 'You cast "Summon Page" and rolled a natural 1.',
  },
  {
    title: 'Potion of Knowledge: Empty',
    description: "The vial's dry; wisdom refuses to pour.",
  },
  {
    title: 'Portal Misaligned',
    description: "Step through here and you'll land in nowhere.",
  },
  {
    title: 'Illusion Dispelled',
    description: 'What looked like a page was only glamour.',
  },
  {
    title: 'Runes Gone Dim',
    description: 'The glyphs that named this path have faded.',
  },

  // Network / Sysadmin (clearly whimsical)
  {
    title: 'Teapot Mode: Activated',
    description: 'We brew leaves, not pages. Try a different pot.',
  },
  {
    title: 'Firewall of Thorns',
    description: 'This route pricked our packets. We turned back.',
  },
  {
    title: 'Cable of Holding: Unplugged',
    description: "The link slipped from reality's socket.",
  },
  {
    title: 'Maintenance Snack Break',
    description: 'Routers nibbling cookies; path not being served.',
  },
  {
    title: 'Packet Turned to Confetti',
    description: 'Arrived festive, unreadable, and everywhere.',
  },

  // Gaming References
  {
    title: 'Loot Chest: Empty',
    description: 'You opened it. Air crit for 404 damage.',
  },
  {
    title: 'Out of Mana',
    description: 'Can\'t cast "Load Page" without more blue bar.',
  },
  {
    title: 'Quest Not Accepted',
    description: 'You skipped the step that unlocks this area.',
  },
  {
    title: 'Boss Room: Key Required',
    description: 'Seek the Skeleton Key of URLs.',
  },
  {
    title: 'Save Slot Missing',
    description: 'Checked every slot. Nothing but dust.',
  },

  // Science / Math
  {
    title: '404 Kelvin',
    description: 'Absolute zero chance of locating this page.',
  },
  {
    title: 'Local Black Hole',
    description: 'Everything nearby got spaghettified, including the link.',
  },
  {
    title: 'Out of Phase',
    description: 'Exists two nanoseconds to the left of now.',
  },
  {
    title: 'String Theory Squeal',
    description: 'Wrong vibration; this universe declined to render.',
  },
  {
    title: 'Butterfly Effect',
    description: 'One pixel fluttered; the page flew away.',
  },

  // Movie / TV
  {
    title: 'Beam Me Up… Wait',
    description: 'Transporter hums, then politely vanishes the page.',
  },
  {
    title: 'One Does Not Simply',
    description: '…walk into this URL without a guide.',
  },
  {
    title: 'Choose Wisely',
    description: 'You picked the cup of dust. Page crumbled accordingly.',
  },
  {
    title: 'Deleted Scene',
    description: "Director's cut trimmed this moment from canon.",
  },
  {
    title: 'Continuity Rebooted',
    description: 'This route exists in the prequel, not the sequel.',
  },

  // Sage / Wizard Phrases
  {
    title: 'The Sage Knows Not This Path',
    description: 'Even the oldest scrolls speak nothing of it.',
  },
  {
    title: 'Omen of Absence',
    description: 'The stars chart an empty place where this should be.',
  },
  {
    title: 'The Candle Burns Low',
    description: 'Seek not this way when the flame falters.',
  },
  {
    title: 'Words Lost to the Wind',
    description: 'Once scribed, now scattered to the directions.',
  },
  {
    title: "The Elder's Silence",
    description: "The oldest among us nod and say: 'Not here.'",
  },

  // Console / Terminal (mock-terminal, obviously playful)
  {
    title: '$ navigate home',
    description: 'Tip: aliases include "/", "root", and "comfort".',
  },
  {
    title: '$ cd ./somewhere-nice',
    description: 'Now leaving /void. Bring a sweater.',
  },
  {
    title: '$ cat this-page.txt',
    description: 'Result: a perfectly blank haiku.',
  },
  {
    title: '$ history | tail -1',
    description: "Last command: 'oops'.",
  },
  {
    title: "$ alias find='ask the menu'",
    description: 'Smart. It knows the good places.',
  },

  // Retro / Vintage
  {
    title: 'Insert Cartridge',
    description: 'Blow gently, reseat, and try another route.',
  },
  {
    title: 'Tape Loading…',
    description: 'ETA: a very nostalgic forever.',
  },
  {
    title: 'Out of Quarters',
    description: 'No credits left to continue this path.',
  },
  {
    title: 'Unsupported Browser Era',
    description: 'Requires Navigator and a time machine.',
  },
  {
    title: 'CRT Afterimage',
    description: 'A ghost of content lingers, unreadable and green.',
  },

  // Random / Absurd
  {
    title: 'Cat Approved Disappearance',
    description: 'Content knocked off the shelf at 3 AM.',
  },
  {
    title: 'Alien Carpool Lane',
    description: 'Page hitched a ride on a passing saucer.',
  },
  {
    title: 'Perpetual Construction',
    description: 'Hardhats on; cones as far as the eye can see.',
  },
  {
    title: 'Invisible Ink',
    description: 'Reveal with moonlight and a patient heart.',
  },
  {
    title: 'Parallel Universe Detour',
    description: 'Your page is thriving—just not here.',
  },
];
