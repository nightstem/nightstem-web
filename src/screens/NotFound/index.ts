export * from '@/screens/NotFound/NotFound';
export { default } from '@/screens/NotFound/NotFound';
