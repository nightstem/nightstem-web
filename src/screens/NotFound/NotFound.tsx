'use client';
import { usePathname, useRouter } from 'next/navigation';
import { useCallback, useEffect, useState } from 'react';

import Logo, {
  LOGO_ANIMATIONS,
  LOGO_MODES,
  LOGO_SIZES,
} from '@/components/ui/Logo';

import randomInt from '@/utils/randomInt';
import hashIndex from '@/utils/hashIndex';

import {
  type NotFoundPhrase,
  defaultNotFound,
  notFoundList,
} from '@/screens/NotFound/constants';
import Kbd from '@/components/ui/Kbd';
import { Button, LinkButton } from '@/components/ui/Buttons';

export type NotFoundProps = {
  notFoundPhrases?: NotFoundPhrase[];
};

const NotFound = ({ notFoundPhrases = notFoundList }: NotFoundProps) => {
  const pathname = usePathname() ?? '/404';
  const router = useRouter();

  const [index, setIndex] = useState(() =>
    hashIndex(pathname, notFoundPhrases.length),
  );
  const shuffle = useCallback(() => {
    if (notFoundPhrases.length <= 1) return;
    let next = randomInt(notFoundPhrases.length);
    if (next === index) next = (next + 1) % notFoundPhrases.length;
    setIndex(next);
  }, [index, notFoundPhrases.length]);

  useEffect(() => {
    const onKey = (e: KeyboardEvent) => {
      if (e.key === 'r' || e.key === 'R') shuffle();
      if (e.key === 'Escape') router.push('/');
    };
    window.addEventListener('keydown', onKey);
    return () => window.removeEventListener('keydown', onKey);
  }, [shuffle, router]);

  const { title, description } = notFoundPhrases[index] ?? defaultNotFound;

  return (
    <main className="screen-container grid h-full w-full place-items-center">
      <h1 className="sr-only">Page not found</h1>

      <section
        aria-labelledby="nf-title"
        aria-describedby="nf-desc"
        className="grid w-full max-w-prose gap-8 px-6"
      >
        <header className="grid gap-2">
          <Logo
            size={LOGO_SIZES.SM}
            mode={LOGO_MODES.HORIZONTAL}
            animation={LOGO_ANIMATIONS.PULSE}
            aria-label="Nightstem"
          />

          <div className="grid gap-8" aria-live="polite" aria-atomic="true">
            <h2 id="nf-title" className="heading-lg">
              {title}
            </h2>

            <p id="nf-desc" className="text-sm text-foreground/60 md:text-base">
              {description}
            </p>
          </div>
        </header>

        <div className="flex flex-col gap-6">
          <div className="flex gap-3">
            <LinkButton href="/">Go home</LinkButton>
            <Button
              onClick={shuffle}
              variant="ghost"
              color="neutral"
              aria-label="Show another message (press R)"
              title="Shuffle (R)"
            >
              Shuffle
            </Button>
          </div>

          <div className="flex items-center gap-1.5 border-t border-foreground/10 pt-3 text-caption text-foreground/60">
            <span>Tip: press</span>
            <Kbd>Esc</Kbd>
            <span>to go home</span>
            <span>â€¢</span>
            <Kbd>R</Kbd>
            <span>to shuffle</span>
          </div>
        </div>
      </section>
    </main>
  );
};

NotFound.displayName = 'NotFound';
export default NotFound;
