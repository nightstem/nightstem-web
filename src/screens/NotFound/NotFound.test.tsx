import userEvent from '@testing-library/user-event';
import { render, screen, act } from '@testing-library/react';
import { usePathname, useRouter } from 'next/navigation';
import { vi, describe, it, expect, beforeEach, afterEach } from 'vitest';
import { axe } from 'vitest-axe';

import NotFound from '@/screens/NotFound/NotFound';
import { notFoundList, defaultNotFound } from '@/screens/NotFound/constants';

vi.mock('next/navigation', () => ({
  usePathname: vi.fn(),
  useRouter: vi.fn(),
}));

const mockPush = vi.fn();
const mockRouter = { push: mockPush } as unknown as ReturnType<
  typeof useRouter
>;

describe('NotFound', () => {
  beforeEach(() => {
    vi.mocked(useRouter).mockReturnValue(mockRouter);
    vi.mocked(usePathname).mockReturnValue('/some-path');
    mockPush.mockClear();
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('renders correctly with snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<NotFound />);
    const results = await act(async () => axe(container!));
    expect(results).toHaveNoViolations();
  });

  describe('Click functionality', () => {
    it('navigates to home when go home link is clicked', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      const goHomeLink = screen.getByRole('link', { name: 'Go home' });
      await user.click(goHomeLink);

      expect(goHomeLink).toHaveAttribute('href', '/');
    });

    it('shuffles content when shuffle button is clicked', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      const shuffleButton = screen.getByRole('button', {
        name: /show another message/i,
      });
      await user.click(shuffleButton);

      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });

    it('shuffles content multiple times to test collision handling', async () => {
      const phrases = [
        { title: 'First', description: 'First desc' },
        { title: 'Second', description: 'Second desc' },
      ];
      const user = userEvent.setup();
      render(<NotFound notFoundPhrases={phrases} />);

      const shuffleButton = screen.getByRole('button', {
        name: /show another message/i,
      });

      for (let i = 0; i < 10; i++) {
        await user.click(shuffleButton);
        expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
      }
    });
  });

  describe('Keyboard functionality', () => {
    it('navigates to home when Escape key is pressed', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      await user.keyboard('{Escape}');

      expect(mockPush).toHaveBeenCalledWith('/');
    });

    it('shuffles content when R key is pressed', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      await user.keyboard('r');

      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });

    it('shuffles content when uppercase R key is pressed', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      await user.keyboard('R');

      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });

    it('ignores other key presses', async () => {
      const user = userEvent.setup();
      render(<NotFound />);

      await user.keyboard('a');
      await user.keyboard('Enter');
      await user.keyboard(' ');

      expect(mockPush).not.toHaveBeenCalled();
      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });
  });

  describe('Edge cases and error scenarios', () => {
    it('handles single phrase array - shuffle should not change content', async () => {
      const singlePhrase = [
        { title: 'Single Title', description: 'Single Description' },
      ];
      const user = userEvent.setup();

      render(<NotFound notFoundPhrases={singlePhrase} />);

      expect(screen.getByText('Single Title')).toBeInTheDocument();
      expect(screen.getByText('Single Description')).toBeInTheDocument();

      const shuffleButton = screen.getByRole('button', {
        name: /show another message/i,
      });
      await user.click(shuffleButton);

      expect(screen.getByText('Single Title')).toBeInTheDocument();
      expect(screen.getByText('Single Description')).toBeInTheDocument();

      await user.keyboard('R');

      expect(screen.getByText('Single Title')).toBeInTheDocument();
      expect(screen.getByText('Single Description')).toBeInTheDocument();
    });

    it('handles undefined pathname', () => {
      vi.mocked(usePathname).mockReturnValue(undefined as unknown as string);

      render(<NotFound />);

      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });

    it('handles null pathname', () => {
      vi.mocked(usePathname).mockReturnValue(null as unknown as string);

      render(<NotFound />);

      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    });

    it('falls back to defaultNotFound when index is out of bounds', () => {
      // Create empty array to force fallback to defaultNotFound
      render(<NotFound notFoundPhrases={[]} />);

      expect(
        screen.getByRole('heading', { level: 2, name: defaultNotFound.title }),
      ).toBeInTheDocument();
      expect(screen.getByText(defaultNotFound.description)).toBeInTheDocument();
    });

    it('handles custom notFoundPhrases prop', () => {
      const customPhrases = [
        { title: 'Custom Title', description: 'Custom Description' },
      ];

      render(<NotFound notFoundPhrases={customPhrases} />);

      expect(screen.getByText('Custom Title')).toBeInTheDocument();
      expect(screen.getByText('Custom Description')).toBeInTheDocument();
    });

    it('has proper ARIA attributes for accessibility', () => {
      render(<NotFound />);

      const section = screen.getByRole('region');
      const heading = screen.getByRole('heading', { level: 2 });

      expect(section).toBeInTheDocument();
      expect(section).toHaveAttribute('aria-labelledby', 'nf-title');
      expect(section).toHaveAttribute('aria-describedby', 'nf-desc');
      expect(heading).toHaveAttribute('id', 'nf-title');
    });

    it('shuffle button has proper accessibility attributes', () => {
      render(<NotFound />);

      const shuffleButton = screen.getByRole('button', {
        name: /show another message/i,
      });

      expect(shuffleButton).toHaveAttribute(
        'aria-label',
        'Show another message (press R)',
      );
      expect(shuffleButton).toHaveAttribute('title', 'Shuffle (R)');
    });
  });

  describe('Content validation', () => {
    it('displays content from notFoundList or default', () => {
      render(<NotFound />);

      const title = screen.getByRole('heading', { level: 2 }).textContent!;
      const description = screen.getByText(/.*/, {
        selector: 'p',
      }).textContent!;

      const isValidContent =
        notFoundList.some(
          (item) => item.title === title && item.description === description,
        ) ||
        (title === defaultNotFound.title &&
          description === defaultNotFound.description);

      expect(isValidContent).toBe(true);
    });

    it('can display different content based on pathname hash', () => {
      vi.mocked(usePathname).mockReturnValue('/path1');
      const { unmount } = render(<NotFound />);
      const title1 = screen.getByRole('heading', { level: 2 }).textContent;
      unmount();

      vi.mocked(usePathname).mockReturnValue('/path2');
      render(<NotFound />);
      const title2 = screen.getByRole('heading', { level: 2 }).textContent;

      expect(title1).toBeTruthy();
      expect(title2).toBeTruthy();
    });
  });
});
