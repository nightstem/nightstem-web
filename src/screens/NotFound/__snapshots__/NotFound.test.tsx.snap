// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > renders correctly with snapshot 1`] = `
<main
  class="h-full w-full grid place-items-center overflow-hidden"
>
  <article
    aria-labelledby="nf-title"
    class="grid gap-6 w-full max-w-[64ch] px-4"
  >
    <header
      class="grid gap-4 w-full"
    >
      <div
        class="inline-flex items-center gap-2"
      >
        <svg
          aria-hidden="true"
          class="text-primary-500 h-[1.066rem] w-[1.066rem] animate-pulse shrink-0"
          fill="none"
          viewBox="0 0 500 500"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M346.04 135.875L500 245.427L161.185 495.252C157.629 497.874 151.828 497.923 148.228 495.362L0.786259 390.448L346.04 135.875Z"
            fill="currentColor"
          />
          <path
            d="M148.322 4.74771C151.878 2.12562 157.68 2.07657 161.279 4.63814L333.785 127.386L179.023 241.499L9.54552e-05 114.113L148.322 4.74771Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="heading-sm"
        >
          Nightstem
        </span>
      </div>
      <h1
        class="heading-lg"
        id="nf-title"
      >
        CRT Afterimage
      </h1>
    </header>
    <div
      class="flex flex-col gap-12"
    >
      <p
        class="text-neutral-400"
      >
        A ghost of content lingers, unreadable and green.
      </p>
      <div
        class="w-full flex flex-col gap-4"
      >
        <div
          class="flex gap-3"
        >
          <button
            class="select-none whitespace-nowrap font-medium cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-dark text-base px-4 py-2 rounded-md text-primary-500 bg-transparent border border-primary-500 hover:border-primary-400 active:border-primary-400 hover:bg-primary-500/20 active:bg-primary-500/20 focus-visible:ring-primary-500 disabled:opacity-60 disabled:hover:border-primary-500 disabled:active:border-primary-500"
            type="button"
          >
            Go home
          </button>
          <button
            aria-label="Show another message (press R)"
            class="select-none whitespace-nowrap font-medium cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-dark text-base px-4 py-2 rounded-md text-neutral-400 bg-transparent hover:text-dark hover:bg-neutral-400/60 active:bg-neutral-400/60 focus-visible:ring-neutral-400 disabled:opacity-60 disabled:hover:text-neutral-400 disabled:hover:bg-transparent disabled:active:bg-transparent"
            title="Shuffle (R)"
            type="button"
          >
            Shuffle
          </button>
        </div>
        <div
          class="flex items-center gap-1.5 text-caption text-neutral-400"
        >
          <span>
            Tip: press
          </span>
          <kbd
            class="rounded shadow-2xl border border-foreground/20 p-1.5"
          >
            Esc
          </kbd>
          <span>
            to go home
          </span>
          <span>
            â€¢
          </span>
          <kbd
            class="rounded shadow-2xl border border-foreground/20 p-1.5"
          >
            R
          </kbd>
          <span>
             to shuffle
          </span>
        </div>
      </div>
    </div>
  </article>
</main>
`;
